src/core/utils/hooks/useContainerWidth.tsx

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useMergedRefs } from './useMergedRefs';
+import { useResizeObserver } from './useResizeObserver';
+
+/**
+ * Hook that returns the width of an element in three stages:
+ *  - initialized with 0
+ *  - immediately set to element's initial width as soon as it's mounted
+ *  - update to new width every time it changes (using `useResizeObserver` hook)
+ *
+ * @private
+ * @param watchResizes If false, ResizeObserver will not be connected and only the initial width will be returned
+ * @returns [ref to attach to the element, stateful width of the element]
+ *
+ * @example
+ * const [ref, width] = useContainerWidth();
+ * ... // do something with width
+ * return <div ref={ref}>...</div>;
+ */
+export const useContainerWidth = <T extends HTMLElement>(
+  watchResizes = true,
```

Comment:
so is there a point to use this hook if you dont want to listen to resizes?? Kinda looks too heavy

Reply:
Yes I think there is. The first callback ref to set initial width is very useful, and there is also the ability to turn off listening to resizes later (we do this in Tabs for default vs rest)

Reply:
What is the use of just initial width? Why would you need to use it, if you can control content width with css (using display inline)

Reply:
well, sometimes you only care about initial width because you want to do something with it (like set a CSS variable)

Reply:
for that you can have just ref and take its width, same 1 liner basically

Reply:
Well that's what I had before, it doesn't work because ref will be null initially
![image](https://user-images.githubusercontent.com/9084735/142450884-fcc20da4-9477-44c9-b6d5-ed47f27f822e.png)

Reply:
oh, I would have expected you use it directly from ref, where you need it.. but yeah